app/controllers/api/v1/workflows_controller.rb

```diff
@@ -81,7 +83,9 @@ def add_relation(resource, relation, value)
   def new_items(resource, relation, value)
     case relation
     when :retired_subjects, 'retired_subjects'
+      resource.save!
       value.flat_map {|id| resource.retire_subject(id) }
+      resource.reload
```

Comment:
does this code even get called when using the `update_links` route? If not we can just enforce `retired_subjects` on a workflow via the update_links route and not the update resource links to avoid workflow instance updates combined with retired counter bumps that cause the lock errors.